libcrux-ml-dsa/src/simd/traits.rs

```diff
@@ -30,6 +30,16 @@ pub(crate) trait Repr: Copy + Clone {
 pub(crate) mod specs {
     use hax_lib::*;
 
+    pub(crate) const PRIME: u32 = 8380417;
+
+    pub(crate) const MONT_R: u32 = 8380417;
+
+    pub(crate) const FIELD_MAX: u32 = 8380416;
+
+    pub(crate) const FIELD_MID: u32 = 4190208;
+
+    pub(crate) const NTT_BASE_BOUND: u32 = FIELD_MID;
+
     type SIMDContent = [i32; COEFFICIENTS_IN_SIMD_UNIT];
     // Avoiding a recursive bundle
```

Comment:
This sounds like it should get a hax bug for improving this? Duplicating stuff like this just because it's in a nested module isn't great.

Reply:
This bit is not a hax bug.  Rust allows circular dependencies between modules but F* does not and we have to deal with it in some way.  hax would correctly inject a recursive bundle here, and I prefer not to have recursive bundles for proof ergonomics, and so we try to avoid trivial circular dependencies in our own code wherever possible, especially when writing specs.